### PR TITLE
Update info about release cycle

### PR DESCRIPTION
**Pull request by drew2a**
This PR updates the document that describes a release cycle.

The goal: to clarify a process. 
Linked to the last dev discussion.

### CODE REVIEW

File: doc/development_methodology.rst [hunk 1]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
...
+
+1. A release branch forks from `devel`
+2. The release branch is checked by running the application-tester
+3. In case no errors, a release candidate (RC) is published as pre-release in Github and published to the forum as release post.
+4. After a while (1-2 weeks) release candidate become a stable
+5. The stable release is delivered to users
+6. The release branch is merged to `devel`
+
+
+In case of an error than needs to be fixed asap in the published stable release:
```
Comment (xoriole) [inline doc/development_methodology.rst:38]: *than -> that

File: doc/development_methodology.rst [hunk 2]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
...
+2. The release branch is checked by running the application-tester
+3. The fixed release is delivered to users
+4. The release branch is merged to `devel`
+
+
+A release is started by forking from the top of **devel**.
+From that moment, all the bugfixes relevant to the current release must be merged into the corresponding release branch.
+No new features could be added to it.
+
+When the release is ready to publishing, it is merged back to **devel** to integrate the bugfixes.
```
Comment (xoriole) [inline doc/development_methodology.rst:50]: Suggestion:
to publishing -> for publishing

**Review by xoriole — CHANGES_REQUESTED** [review]

File: doc/development_methodology.rst [hunk 3]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
 
 Release lifecycle
 -----------------
-A release is started by forking from the top of **devel**. The first commit added to the release branch bumps the release version. From that moment, all the bugfixes relevant to the current release must be merged only into the corresponding release branch. No new features could be added to it.
 
-When the release is ready, it is merged into **master** with the suitable release tag. Next it is merged into the current **devel** branch to integrate the bugfixes. If a bugfix from the current release branch is really, really necessary for the current **devel** branch, it could be cherry-picked onto **devel**. But be aware that each of these cherry-pickings eventually results in a merge conflict that should be resolved manually.
+We follows `SemVer <https://semver.org/>`_ notation for release naming.
```
Comment (xoriole) [inline doc/development_methodology.rst:23]: Suggestion:
follows -> follow

File: doc/development_methodology.rst [hunk 4]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
...
 
-The release branch lives in the Tribler repository for as long as we support the corresponding Tribler version. Eventually, the branch is removed.
+* Stable release name: <MAJOR>.<MINOR>.<PATCH>
+* Release candidate name: <MAJOR>.<MINOR>.<PATCH>-RC<NUMBER>
+
+The release lifecycle:
+
+1. A release branch forks from `devel`
```
[the remark this answers is not in the document]
Comment (drew2a) [reply doc/development_methodology.rst]: Has added in the chapter above.

File: doc/development_methodology.rst [hunk 5]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
...
+2. The release branch is checked by running the application-tester
+3. In case no errors, a release candidate (RC) is published as pre-release in Github and published to the forum as release post.
+4. After a while (1-2 weeks) release candidate become a stable
```
[the remark this answers is not in the document]
Comment (drew2a) [reply doc/development_methodology.rst]: Fixed

File: doc/development_methodology.rst [hunk 6]
```diff
@@ -19,11 +19,35 @@ Our main repository contains 2 permanent branches:
...
+5. The stable release is delivered to users
```
[the remark this answers is not in the document]
Comment (drew2a) [reply doc/development_methodology.rst]: Fixed

File: doc/development_methodology.rst (hunk 2 above)
Comment (drew2a) [reply doc/development_methodology.rst]: Fixed

File: doc/development_methodology.rst (hunk 3 above)
Comment (drew2a) [reply doc/development_methodology.rst]: Fixed

**Comment by drew2a** [comment]
@xoriole thank you for your comments and suggestions.

File: doc/development_methodology.rst (hunk 1 above)
Comment (drew2a) [reply doc/development_methodology.rst]: fixed

**Review by xoriole — APPROVED** [review]

**Review by kozlovsky — APPROVED** [review]

**Pull request merged**